### PR TITLE
Backport 2104 3-0-stable

### PR DESCRIPTION
**Pull request by JoeDupuis**
Bug fix #2104 backported to 3-0-stable (to cut a release).

Follow up from #2163

### CODE REVIEW

**Review by ioquatix — APPROVED** [review]

File: lib/rack/multipart/parser.rb [hunk 1]
```diff
@@ -213,6 +213,8 @@ def initialize(boundary, tempfile, bufsize, query_parser)
 
         @sbuf = StringScanner.new("".dup)
         @body_regex = /(?:#{EOL}|\A)--#{Regexp.quote(boundary)}(?:#{EOL}|--)/m
+        @body_regex_at_end = /#{@body_regex}\z/m
```
Comment (jeremyevans) [inline lib/rack/multipart/parser.rb:216]: I don't think `@body_regex_at_end` is used by this code, so it should be added.

**Review by jeremyevans — CHANGES_REQUESTED** [review]
One minor change requested, then this should be good.

File: lib/rack/multipart/parser.rb (hunk 1 above)
Comment (JoeDupuis) [reply lib/rack/multipart/parser.rb]: Oops, it came with the cherry pick by accident. I removed it.

**Review by jeremyevans — APPROVED** [review]

**Comment by ioquatix** [comment]
Thanks everyone!

**Pull request merged**